ci-runner.py

```diff
@@ -1,90 +1,131 @@
+        'PERSISTENT_MONITORING':  os.getenv('PERSISTENT_MONITORING', 'false'),
+    }
+    for key in default_env_vars:
+        try:
+            if key == 'SKIP_STEPS':
+                SKIP_STEPS = os.getenv('SKIP_STEPS').split(',')
+                print(f"Setting key={key} value={SKIP_STEPS}")
+                continue
+            print(f"Setting key={key} value={default_env_vars[key]}")
+            os.environ[key] = default_env_vars[key]
+        except Exception as e:
+            logging.error(f"Error setting environment variable {key}: {e}")
+
+def load_env_ref(envs):
+    """
+    Set default environment variables from ref (step) definition.
+    """
+    print("Setting up environment variables for ref")
+    for env_def in envs:
+        try:
+            # this allow override of env vars from ref
+            if env_def['name'] not in os.environ:
+                print(f"Setting default env value from Ref definition: key={env_def['name']} value={env_def['default']}")
+                os.environ[env_def['name']] = env_def['default']
+        except Exception as e:
+            logging.error(f"Error setting environment variable {env_def['name']}: {e}")
+
 
 def initialize():
+    """
+    Initialize the local directories to prepare for running steps.
+    """
     print("Initializing")
-    if os.path.exists(f'/tmp/{env.CLUSTER_NAME}'):
-        shutil.rmtree(f'/tmp/{env.CLUSTER_NAME}')
+    if os.path.exists(f'/tmp/{os.getenv('.CLUSTER_NAME')}'):
+        shutil.rmtree(f'/tmp/{os.getenv('.CLUSTER_NAME')}')
     
     if os.path.exists(f'/tmp/install-dir'):
         shutil.rmtree('/tmp/install-dir')
 
     print("Setting up required artifacts")
-    os.mkdir(f'/tmp/{env.CLUSTER_NAME}')
-    os.mkdir(f'/tmp/{env.CLUSTER_NAME}/shared')
-    os.mkdir(f'/tmp/{env.CLUSTER_NAME}/profile')
-    os.mkdir(f'/tmp/{env.CLUSTER_NAME}/artifact')
-    shutil.copyfile(env.SSH_PUBLIC_KEY, f'{env.PROFILE_DIR}/ssh-publickey')
-    shutil.copyfile(env.PULL_SECRET_FILE, f'{env.PROFILE_DIR}/pull-secret')
-    shutil.copyfile(env.AWS_CREDENTIAL_PATH, f'{env.PROFILE_DIR}/.awscred')
-
-def scan_dir_for(type, path, name, pathParts=""):        
-    if os.path.isfile(path):
-        filename = os.path.basename(path)
-        if filename == name + "-" + type + ".yaml":
-            return path
-        else:
-            return 
-
-    for filename in os.listdir(path):        
-        if os.path.isfile(filename):
-            pass
-        if pathParts != "":
-            thisPath = filename
-        else:
-            thisPath = "-" + filename
-        foundPath = scan_dir_for(type, path + "/" + filename, name, thisPath)
-        if foundPath != None:
-            return foundPath
+    os.mkdir(f'/tmp/{os.getenv('CLUSTER_NAME')}')
+    os.mkdir(f'/tmp/{os.getenv('CLUSTER_NAME')}/shared')
+    os.mkdir(f'/tmp/{os.getenv('CLUSTER_NAME')}/profile')
+    os.mkdir(f'/tmp/{os.getenv('CLUSTER_NAME')}/artifact')
```

Comment:
those are vars, wondering if we can use the var instead of mounting the path here